labs/a3.md

```diff
@@ -0,0 +1,520 @@
+> for more info. Also, notice that we are already connected to the
+> internet! Our virtual network uses DHCP, and the archiso knows to
+> try to autoconfigure using DHCP. We will talk more about DHCP and
+> other networking configuration later in the course. Finally, notice
+> that we've conveniently inherited the system clock, so the time
+> inside the VM should be right. You can run `date` to see.
+
+## partitioning time
+
+The first order of business is to set up the partitions that we will
+be using for this installation. We are going to use a simple parition
+scheme:
+- an EFI system partition (ESP)
+- one encrypted partition that holds our root filesystem
+- a small partition for swap space
+
+This is a very common scheme and is totally a scheme you could use if
+you were installing Arch on your own computer (except that you'll
+hopefully have more than 5GiB to work with)!
+
+### create partitions
+
+We will use `fdisk` to create our partitions. Run `fdisk /dev/sda` to start.
+
+> Note that you'll be making a bunch of changes to the parition table,
+> but these changes are not made immediately. So, if you make a
+> mistake, you can just start again from step 1 without damaging
+> anything.
+
+1. Create a GPT table by typing `g` (then hit enter).
+2. Create your ESP by typing `n`. Use the default parition number and
+   first sector (just hit enter to automatically take the default).
+   For the last sector, type `+512M` to allocate 512MiB for the
+   partition.
+3. Type `t` to change the type of the new partition (parition 1) to
+   "EFI System". You will have to list the types and find the right
+   ID. (Note: press `q` to get out of the list of filesystem types).
+4. Create your main root partition by typing `n` again. Use the
+   default partition number and first sector. To leave 512MiB at the
+   end of the disk for our swap partition, type `-512M` for the last
+   sector. (This partition should already have the correct type,
+   "Linux filesystem", so you won't need to change that.)
+5. Create your swap partition by typing `n` one last time, and just
+   take all the defaults.
+6. Type `t` to change the type of the swap partition (parition 3) to
+   "Linux swap", as in step 3.
+7. Type `p` to show the current partition table with all the changes
+   you've made. You should see an "EFI System" partition that is
+   512MiB, a "Linux filesystem" partition that is 4GiB, and a "Linux
+   swap" partition that is 512MiB.
+8. If everything looks right, type `w` to finally actually write the
+   changes to disk.
+
+At this point, `fdisk` should exit.
+
+**GRADESCOPE**: To see your final results, run `fdisk -l /dev/sda` and
+paste the output into Gradescope.
+
+At this point, you've created the disk partitions, but nothing is on
+them yet. Now we need to put the filesystems you will use on these
+partitions!
+
+### create filesystems
+
+#### ESP
+
+The ESP (EFI System Partition) has to be in a FAT format. For more
```

Comment:
Link to filesystems

Reply:
wym?

Reply:
I really should've read the lecture first, but will the phrase "FAT format" be coherent to the students?